apollo-router/src/configuration/metrics.rs

```diff
@@ -336,20 +321,74 @@ impl Metrics {
             "$..logging.experimental_when_header"
         );
 
-        log_usage_metrics!(
-            value.apollo.router.config.batching,
+        populate_config_instrument!(
+            apollo.router.config.batching,
             "$.experimental_batching[?(@.enabled == true)]",
             opt.mode,
             "$.mode"
         );
     }
+
+    fn populate_env_instrument(&mut self) {
+        #[cfg(not(test))]
+        fn env_var_exists(env_name: &str) -> opentelemetry::Value {
+            std::env::var(env_name)
+                .map(|_| true)
+                .unwrap_or(false)
+                .into()
+        }
+        #[cfg(test)]
+        fn env_var_exists(_env_name: &str) -> opentelemetry::Value {
+            true.into()
+        }
+
+        let mut attributes = HashMap::new();
+        attributes.insert("env.apollo.key".to_string(), env_var_exists("APOLLO_KEY"));
+        attributes.insert(
+            "env.apollo.graph_ref".to_string(),
+            env_var_exists("APOLLO_GRAPH_REF"),
+        );
+        attributes.insert(
+            "env.apollo.license".to_string(),
+            env_var_exists("APOLLO_ROUTER_LICENSE"),
+        );
+        attributes.insert(
+            "env.apollo.license.path".to_string(),
+            env_var_exists("APOLLO_ROUTER_LICENSE_PATH"),
+        );
+
+        self.data
+            .insert("apollo.router.env".to_string(), (1, attributes));
+    }
+}
+impl From<InstrumentData> for Metrics {
+    fn from(data: InstrumentData) -> Self {
+        Metrics {
+            _instruments: data
+                .data
+                .into_iter()
+                .map(|(metric_name, (value, attributes))| {
+                    let attributes: Vec<_> = attributes
+                        .into_iter()
+                        .map(|(k, v)| KeyValue::new(k.replace("__", "."), v.clone()))
```

Comment:
```suggestion
                        .map(|(k, v)| KeyValue::new(k.replace("__", "."), v))
```

is the clone necessary here?

Reply:
[a4e3605](https://github.com/apollographql/router/pull/4364/commits/a4e36056787230e0182223ffb78d9126e362dca1)